src/Stratis.Bitcoin.Features.Consensus/Checkpoints.cs

```diff
@@ -139,9 +140,11 @@ public class Checkpoints : ICheckpoints
         /// Initializes a new instance of the object.
         /// </summary>
         /// <param name="network">Specification of the network the node runs on - regtest/testnet/mainnet/stratis test/main.</param>
-        public Checkpoints(Network network)
+        /// <param name="settings">Settings for full node - used to see if checkpoints have been disabled or not.</param>
+        public Checkpoints(Network network, NodeSettings settings)
         {
-            if (network.Equals(Network.Main)) this.checkpoints = bitcoinMainnetCheckpoints;
+            if (settings != null && !settings.UseCheckpoints) this.checkpoints = new Dictionary<int, CheckpointInfo>(); 
```

Comment:
if you want to make me happy, pls use parentheses like this `if ((settings != null) && !settings.UseCheckpoints)` - i.e. use them even if not needed, so that one does not need to know the operator priority